insights/parsers/kpatch_patches.py

```diff
@@ -0,0 +1,40 @@
+"""
+KpatchPatches - report locally stored kpatch patches
+====================================================
+
+This parser creates a list of the module names of locally
+stored kpatch modules. If no modules are installed, the
+list will be empty.
+
+"""
+
+from .. import parser, Parser
+from insights.specs import Specs
+import re
+
+
+@parser(Specs.kpatch_patch_files)
+class KpatchPatches(Parser):
+    """
+    A parser for getting locally stored kpatch-patch modules
+    """
+    __re_cannot_access = re.compile(r"No such file or directory$")
+    __re_archive = re.compile(r"\.xz$")
+
+    def get_patches(self):
+        return self.patches
+
+    def parse_content(self, content):
+        self.patches = content
+        # If kpatch is not installed, the directory will not exist.
+        # In that case, the list of local kpatch files is empty.
+        if self.__re_cannot_access.search(content[0]):
+            self.patches = []
+            return
+
+        # remove module archives, convert dashes to underscores, remove ".ko" suffixes
+        self.patches =  \
+            list(map(lambda x: x.replace(".ko", ""),
+                list(map(lambda x: x.replace("-", "_"),
```

Comment:
I just wonder why here replace the `-` to `_`? After this, it's difficult to get the original name of the patches. E.g.

In the test case, there is `asdfasdfasdf_asdfasdfasdf-asdfasdfasdf_asdfasdfasdf.ko`, but after this `replace`, we'll get `asdfasdfasdf_asdfasdfasdf_asdfasdfasdf_asdfasdfasdf`. from the result, we don't know what the real original name is.
And the following 3 different names, will result in the same
```
asdfasdfasdf-asdfasdfasdf-asdfasdfasdf-asdfasdfasdf.ko
asdfasdfasdf-asdfasdfasdf_asdfasdfasdf-asdfasdfasdf.ko
asdfasdfasdf-asdfasdfasdf_asdfasdfasdf_asdfasdfasdf.ko
```
Do you think so?

Reply:
The idea here is to produce a list of module names corresponding to the files in the /var/lib/kpatch/$(uname -r) directory that one could expect to find in the output of lsmod. While you do raise a valid point, I have spoken with some of the kpatch developers and they told me that the patch files will include kernel version information in the filename, so name collisions such as the above will be avoided. However, the filenames will include a combination of dashes and underscores so I test that these are all converted to underscores.

Reply:
Looks good to me. Thanks for the explanation.